extinguish.lua

```diff
@@ -0,0 +1,187 @@
+-- Puts out fires.
+-- author: Atomic Chicken
+
+local utils = require 'utils'
+local validArgs = utils.invert({
+'all',
+'location',
+'help'
+})
+local args = utils.processArgs({...}, validArgs)
+
+--@ module = true
+
+local usage = [====[
+
+extinguish
+==========
+This script allows the user to put out fires affecting
+map tiles, plants, units, items and buildings.
+
+To select a target, place the cursor over it before running the script.
+Alternatively, use one of the arguments below.
+
+Arguments::
+
+    -all
+        extinguish everything on the map
+
+    -location [ x y z ]
+        extinguish only at the specified coordinates
+]====]
+
+if args.help then
```

Comment:
The 1-space indentation here should be changed to 2 spaces to match the rest of the file